src/Microsoft.PowerShell.Commands.Utility/commands/utility/FormatAndOutput/format-hex/Format-Hex.cs

```diff
@@ -0,0 +1,334 @@
+                            WriteError(errorRecord);
+                            continue;
+                        }
+                    }
+                }
+
+                if (!provider.Name.Equals("FileSystem", StringComparison.OrdinalIgnoreCase))
+                {
+                    // Write a non-terminating error message indicating that path specified is not supported.
+                    string errorMessage = StringUtil.Format(UtilityCommonStrings.FormatHexOnlySupportsFileSystemPaths, currentPath);
+                    ErrorRecord errorRecord = new ErrorRecord(new ArgumentException(errorMessage),
+                                                                "FormatHexOnlySupportsFileSystemPaths",
+                                                                ErrorCategory.InvalidArgument,
+                                                                currentPath);
+                    WriteError(errorRecord);
+                    continue;
+                }
+
+                pathsToProcess.AddRange(newPaths);
+            }
+
+            return pathsToProcess;
+        }
+
+        /// <summary>
+        /// Pass each valid path on to process its contents.
+        /// </summary>
+        /// <param name="pathsToProcess"></param>
+        private void ProcessPath(List<string> pathsToProcess)
+        {
+            // If there are no paths to proccess, just return.
+            if (pathsToProcess.Count == 0)
+            {
+                return;
+            }
+
+            foreach (string path in pathsToProcess)
+            {
+                ProcessFileContent(path);
+            }
+        }
+
+        /// <summary>
+        /// Creates a binary reader that reads the file content into a buffer (byte[]) 16 bytes at a time, and
+        /// passes a copy of that array on to the ConvertToHexidecimal method to output.
+        /// </summary>
+        /// <param name="path"></param>
+        private void ProcessFileContent(string path)
+        {
+            byte[] buffer = new byte[BUFFERSIZE];
+
+            try
+            {
+                using (BinaryReader reader = new BinaryReader(File.Open(path, FileMode.Open, FileAccess.Read)))
+                {
+                    UInt32 offset = 0;
+                    Int32 bytesRead = 0;
+
+                    while ((bytesRead = reader.Read(buffer, 0, BUFFERSIZE)) > 0)
+                    {
+                        if (bytesRead == BUFFERSIZE)
+                        {
+                            // We are reusing the same buffer so if we save the output to a variable, the variable
+                            // will just contain multiple references to the same buffer memory space (containing only the
+                            // last bytes of the file read).  Cloning the buffer allows us to pass the values on without
+                            // overwriting previous values.
+                            byte[] bufferClone = (byte[])buffer.Clone();
```

Comment:
I've changed it to copy the buffer into a new array instead of cloning the buffer.  If this isn't what you had in mind, please file an issue and @Francisco-Gamino  will follow up.  Due to time constraints with my internship, I won't be able to follow up after today.